lightning/src/ln/channel.rs

```diff
@@ -3179,18 +3171,36 @@ impl<Signer: WriteableEcdsaChannelSigner> Channel<Signer> {
 				need_commitment = true;
 			}
 		}
+		let mut claimed_htlcs = Vec::new();
 		for htlc in self.pending_outbound_htlcs.iter_mut() {
 			if let &mut OutboundHTLCState::RemoteRemoved(ref mut outcome) = &mut htlc.state {
 				log_trace!(logger, "Updating HTLC {} to AwaitingRemoteRevokeToRemove due to commitment_signed in channel {}.",
 					log_bytes!(htlc.payment_hash.0), log_bytes!(self.channel_id));
 				// Grab the preimage, if it exists, instead of cloning
 				let mut reason = OutboundHTLCOutcome::Success(None);
 				mem::swap(outcome, &mut reason);
+				if let OutboundHTLCOutcome::Success(Some(preimage)) = reason {
+					// While failing to handle the `Success(None)` case here could result in
+					// forgetting some HTLC claims, it is only reachable if the HTLC was claimed on
+					// LDK 0.0.104 and prior versions, then upgrading to a modern LDK before the
```

Comment:
This sentence is really confusing, especially the last part `, then`

Reply:
Honestly no — I have hard times understanding the construction "While X, it is only Y, then Z".

Especially I'm getting confused to interpreting the Z part, I genuinely can't understand what's the predicate part of this `then`. Maybe the commas placing, I don't know.

This is certainly a combination of my non-native English and my poor code understanding. If my description above didn't help, feel free to resolve my comment without addressing it.